modules/node_modules/@frogpond/ccci-stolaf-college/v1/jobs/index.js

```diff
@@ -1,52 +1,85 @@
 import {get, ONE_DAY} from '@frogpond/ccc-lib'
 import mem from 'mem'
+import pMap from 'p-map'
 import _jsdom from 'jsdom'
 import getUrls from 'get-urls'
 
 const {JSDOM} = _jsdom
 
-export function cleanJob(job) {
-	// these all need to retain their newlines
-	const description = cleanTextBlock(
-		JSDOM.fragment(job.description).textContent,
-	)
-	const comments = cleanTextBlock(JSDOM.fragment(job.comments).textContent)
-	const skills = cleanTextBlock(JSDOM.fragment(job.skills).textContent)
-	const howToApply = cleanTextBlock(JSDOM.fragment(job.howToApply).textContent)
-	const timeline = cleanTextBlock(JSDOM.fragment(job.timeline).textContent)
-	const timeOfHours = cleanTextBlock(
-		JSDOM.fragment(job.timeOfHours).textContent,
-	)
-
-	const contactEmail = fixupEmailFormat(job.contactEmail)
-	const contactPhone = fixupPhoneFormat(job.contactPhone)
+const jobsUrl =
+	'https://wp.stolaf.edu/student-jobs/wp-json/wp/v2/pages/80'
+
+const PARAGRAPHICAL_KEYS = ['Job Description', 'Skills Needed', 'Additional Comments', 'How to Apply', 'Hiring Timeline']
+
+async function fetchDetail(url) {
+	const resp = await get(url)
+
+	// run-scripts value is needed to properly evaluate javascript to display an email address.
+	// see the jsdom documentation for more details https://github.com/jsdom/jsdom#executing-scripts
+	const dom = new JSDOM(resp.body, {contentType: 'text/html', runScripts: 'dangerously' })
+
+	const id = url.replace(/\D/g,'')
+	const titleText = dom.window.document.querySelector('.gv-list-view-title > h3').textContent
+	const details = dom.window.document.querySelectorAll('div')
+
+	const detailMap = [...details].reduce((coll, listEl) => {
+		let [key, ...value] = listEl.childNodes
+		key = key ? key.textContent.replace(/:$/, '') : key
+
+		if (PARAGRAPHICAL_KEYS.includes(key)) {
+			let paragraphs = [...listEl.querySelectorAll('p')]
+			let content = paragraphs.length ? paragraphs : value
+			value = content
+				.map((el) => el.textContent)
+				.join('\n\n')
+				.trim()
+		} else {
+			value = value
+				.map((el) => el.textContent)
+				.join(' ')
+				.trim()
+		}
+		coll.set(key, value)
+
+		return coll
+	}, new Map())
```

Comment:
There is a lot of logic packed into an anonymous function. Can this be broken out into a function and given a name? I am not really seeing any interaction with the outer scope so I think it should be a simple extraction.

Reply:
It's currently tightly coupled due to how I implemented the helper function that performs a lookup/get on a string. I was trying to avoid passing in a reference to the larger parsed structure on each call.

Anyways. I agree that the complexity of this needs to be broken out and refactored.